File: strong.py
import math

def is_strong_number(num):
    """
    Checks if a given number is a strong number.
    A strong number is a number where the sum of the factorial of its digits
    equals the number itself.
    """
    original_num = num
    sum_of_factorials = 0
    
    # Handle the case of 0 separately as 0! is 1.
    if num == 0:
        return False # Strong numbers are typically positive integers.

    while num > 0:
        digit = num % 10
        sum_of_factorials += math.factorial(digit)
        num //= 10
    
    return sum_of_factorials == original_num

def find_strong_numbers_in_range(start, end):
    """
    Finds all strong numbers within a specified range (inclusive).
    """
    strong_numbers = []
    for i in range(start, end + 1):
        if is_strong_number(i):
            strong_numbers.append(i)
    return strong_numbers

# Find strong numbers between 1 and 5000
strong_numbers_found = find_strong_numbers_in_range(1, 5000)
print(f"Strong numbers between 1 and 5000 are: {strong_numbers_found}")

File: strong.java
public class strong {

    // Pre-computed factorials for digits 0-9 for efficiency
    private static final int[] FACTORIALS = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880};

    /**
     * Checks if a given number is a strong number.
     * A strong number is a number whose sum of the factorial of its digits equals the original number.
     *
     * @param number The number to check.
     * @return true if the number is a strong number, false otherwise.
     */
    public static boolean isStrong(int number) {
        if (number < 0) { // Strong numbers are typically defined for non-negative integers
            return false;
        }
        int originalNumber = number;
        int sumOfFactorials = 0;

        while (number > 0) {
            int digit = number % 10;
            sumOfFactorials += FACTORIALS[digit];
            number /= 10;
        }

        return sumOfFactorials == originalNumber;
    }

    public static void main(String[] args) {
        System.out.println("Strong numbers from 1 to 5000:");
        for (int i = 1; i <= 5000; i++) {
            if (isStrong(i)) {
                System.out.println(i);
            }
        }
    }
}